src/amplitude-client.js

```diff
@@ -342,7 +345,9 @@ var _parseConfig = function _parseConfig(options, config) {
   };
 
   for (var key in config) {
-    if (Object.prototype.hasOwnProperty.call(config, key)) {
+    if (freeFormObjectKeys.has(key)) {
+      options[key] = { ...options[key], ...config[key] };
```

Comment:
I think you might want to add a TODO to find a better soln here - this bypasses **all** checks, right? 

There are still probably some checks to this - as in, header values can't be functions or nested objects, right?